src/Microsoft.ApplicationInsights.AspNet/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -65,19 +59,73 @@ public static void AddApplicationInsightsTelemetry(this IServiceCollection servi
+            };
+            configurationSourceRoot.Add(configSource);
+
+            return configurationSourceRoot;
+        }
+
+        /// <summary>
+        /// Read from configuration
+        /// Config.json will look like this:
+        ///
+        ///      "ApplicationInsights": {
+        ///          "InstrumentationKey": "11111111-2222-3333-4444-555555555555"
+        ///          "TelemetryChannel": {
+        ///              EndpointAddress: "http://dc.services.visualstudio.com/v2/track",
+        ///              DeveloperMode: true
+        ///          }
+        ///      }
+        /// Values can also be read from environment variables to support azure web sites configuration:
+        /// </summary>
+        /// <param name="config">Configuration to read variables from.</param>
+        /// <param name="telemetryConfiguration">Telemetry configuration to populate</param>
+        private static void AddTelemetryConfiguration(IConfiguration config, TelemetryConfiguration telemetryConfiguration)
         {
-            // Read from configuration
-            // Config.json will look like this:
-            //
-            //      "ApplicationInsights": {
-            //            "InstrumentationKey": "11111111-2222-3333-4444-555555555555"
-            //      }
-            string instrumentationKey = config.Get("ApplicationInsights:InstrumentationKey");
+            string instrumentationKey = config.Get(ConfigurationConstants.InstrumentationKeyForWebSites);
+            if (string.IsNullOrWhiteSpace(instrumentationKey))
+            {
+                instrumentationKey = config.Get(ConfigurationConstants.InstrumentationKeyFromConfig);
+            }
             if (!string.IsNullOrWhiteSpace(instrumentationKey))
             {
                 telemetryConfiguration.InstrumentationKey = instrumentationKey;
             }
+
+            string developerModeValue = config.Get(ConfigurationConstants.DeveloperModeForWebSites);
+            if (string.IsNullOrWhiteSpace(developerModeValue))
+            {
+                developerModeValue = config.Get(ConfigurationConstants.DeveloperModeFromConfig);
+            }
+            if (!string.IsNullOrWhiteSpace(developerModeValue))
+            {
+                bool developerMode = false;
+                if (bool.TryParse(developerModeValue, out developerMode))
+                {
+                    telemetryConfiguration.TelemetryChannel.DeveloperMode = developerMode;
+                }
+            }
+
+            string endpointAddress = config.Get(ConfigurationConstants.EndpointAddressForWebSites);
+            if (string.IsNullOrWhiteSpace(endpointAddress))
+            {
+                endpointAddress = config.Get(ConfigurationConstants.EndpointAddressFromConfig);
+            }
+            if (!string.IsNullOrWhiteSpace(endpointAddress))
+            {
+                var channel = telemetryConfiguration.TelemetryChannel as InProcessTelemetryChannel;
```

Comment:
EndpointAddress is a property of ITelemetryChannel. Why are we hardcoding a specific channel type here?